File: Leetcode 1496. Path Crossing/lc 1496.java
class Solution {
    public boolean isPathCrossing(String path) {
        HashSet<String> s=new HashSet<>();
        s.add("0,0");
        int x=0;
        int y=0;
        for(char c:path.toCharArray()){
            if(c=='N')y++;
            else if(c=='S')y--;
            else if(c=='E')x++;
            else x--;

            String cord=x+","+y;
            if(s.contains(cord))return true;
            s.add(cord);
        }
        return false;
    }
}

File: Leetcode 1496. Path Crossing/lc 1496.cpp
class Solution {
public:
    bool isPathCrossing(string path) {
        unordered_set<string> s;
        s.insert({"0,0"});
        int x=0;
        int y=0;
        for(char c:path){
            if(c=='N')y++;
            else if(c=='S')y--;
            else if(c=='E')x++;
            else x--;

            string cord=to_string(x)+","+to_string(y);
            if(s.find(cord)!=s.end())return true;
            s.insert(cord);
        }
        return false;
    }
};